### PR TITLE
Add workload executor for the synchronous Java driver

### PR DESCRIPTION
**Pull request by jyemin**
Probably just one reviewer is sufficient, but not sure who is should be.  Feel free to choose one and remove the other.

Patch build [here](https://evergreen.mongodb.com/build/drivers_atlas_testing_tests_java__driver~java_master_platform~ubuntu_18.04_runtime~java11_patch_86b7d46cdc738755009edbc2ca08b41d85d8d2f5_6050d98c850e61097b2f157c_21_03_16_16_15_22)

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -322,10 +322,10 @@ axes:
       - id: java-master
         display_name: "Java (master)"
         variables:
-          DRIVER_DIRNAME: "java"
+          DRIVER_DIRNAME: "java/sync"
```
Comment (jyemin) [inline .evergreen/config.yml:325]: Eventually we may want to test the reactive streams driver, so leaving room for that here.

File: .evergreen/config.yml [hunk 2]
```diff
@@ -322,10 +322,10 @@ axes:
...
           DRIVER_REPOSITORY: "https://github.com/mongodb/mongo-java-driver"
           DRIVER_REVISION: "master"
-          ASTROLABE_EXECUTOR_STARTUP_TIME: 15
+          ASTROLABE_EXECUTOR_STARTUP_TIME: 3
```
Comment (jyemin) [inline .evergreen/config.yml:328]: Startup is much faster now that gradle is no longer used to run the executor

File: integrations/java/sync/workload-executor [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -o errexit
+
+workloadExecutorFile="/tmp/java-driver-workload-executor.json"
+
+rm -f "$workloadExecutorFile"
+
+# cat the workload executor JSON to a file
```
Comment (jyemin) [inline integrations/java/sync/workload-executor:9]: It's easier to test the script in an IDE if the workload is passed indirectly as a file.  That's why we're not just passing the whole string as a command line parameter

File: integrations/java/sync/workload-executor [hunk 4]
```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -o errexit
+
+workloadExecutorFile="/tmp/java-driver-workload-executor.json"
```
Comment (p-mongo) [inline integrations/java/sync/workload-executor:5]: Normally `mktemp` would be used instead of hardcoding the path like this.

**Review by p-mongo — APPROVED** [review]

**Pull request merged**